src/us/kbase/auth2/lib/Password.java

```diff
@@ -10,21 +18,32 @@
  * well.
  * 
  * @author gaprice@lbl.gov
+ * @author mwsneddon@lbl.gov
  *
  */
 public class Password {
 
+	/** Sets the minimum strength score required of a password.  The zxcvbn strength score is:
+	 * 
+	 * (from https://github.com/dropbox/zxcvbn, https://github.com/nulab/zxcvbn4j)
+	 *   Integer from 0-4 (useful for implementing a strength bar)
+	 *   0 # too guessable: risky password. (guesses < 10^3)
+	 *   1 # very guessable: protection from throttled online attacks. (guesses < 10^6)
+	 *   2 # somewhat guessable: protection from unthrottled online attacks. (guesses < 10^8)
+	 *   3 # safely unguessable: moderate protection from offline slow-hash scenario. (guesses < 10^10)
+	 *   4 # very unguessable: strong protection from offline slow-hash scenario. (guesses >= 10^10)
+	 */
+	private static final int MIN_PASSWORD_STRENGTH_SCORE = 3;
+	
+	private static final int MAX_PASSWORD_LENGTH = 256;
+	
 	private final char[] password;
 	
 	/** Create a password. Note that the incoming array is not copied, and any changes to the array
 	 * will be reflected in this class.
 	 * @param password the password to wrap.
 	 */
 	public Password(final char[] password) {
-		//TODO PWD appropriate checking that the password is strong
-		//TODO PWD max length
-		// https://github.com/nulab/zxcvbn4j
-		// may need to have a non-checking constructor, only need to check on creation
```

Comment:
Hmm, I'm thinking we should make a copy of the array here so that the array can't be modified after the validity check. Similarly a copy should be returned in the getPassword() method. The documentation should note this and that the user should zero out the source array and any arrays from getPassword() ASAP.

We could make a static clear() method to help out here.